apps/design/backend/src/app.ts

```diff
@@ -146,6 +148,90 @@ export function convertVxfPrecincts(election: Election): SplittablePrecinct[] {
   });
 }
 
+function regenerateIds(
+  election: Election,
+  precincts: SplittablePrecinct[]
+): {
+  districts: District[];
+  precincts: SplittablePrecinct[];
+  parties: Party[];
+  contests: AnyContest[];
+} {
+  const idMap = new Map<string, string>();
+  function replaceId<T extends string>(id: T): T {
+    if (!idMap.has(id)) {
+      idMap.set(id, generateId());
```

Comment:
Looks like a fair amount of synchronous crypto happening here - is this is safe for the main thread? Wondering if we should use an async version of the ID generator (probably fine given the number of elements and the size of the IDs, but just double-checking)

Reply:
Good q, I didn't think about that!

It looks like it's pretty fast according to the [benchmarks](https://github.com/ai/nanoid?tab=readme-ov-file#benchmark) (and we're also using a shorter ID length than the default).

There is also a [non-secure option](https://github.com/ai/nanoid?tab=readme-ov-file#non-secure) that is faster that we could consider. Or making it async would be fine too.

I'm also thinking about how rare it is to clone/load an election, so that also makes me less worried.